Add pg port to specified pgaddr without port

nak3 opened the pull request:
<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3911)

<!-- Reviewable:end -->

nak3 committed:
Add pg port to specified pgaddr without port

petermattis commented on sql/pgwire/server.go:
```diff
@@ -36,6 +37,8 @@ const (
 	versionSSL = 80877103
 )
 
+const pgPort = 15432
```
Seems a bit strange that we're duplicating the port here and in `server/context.go`. Can we move this logic into `server` near the usage of `context.PGAddr`? And why is this necessary for `context.PGAddr` but not `context.Addr`? Or maybe it is necessary for `context.Addr` but I'm just not seeing where it is done.

nak3 commented:
Thank you @petermattis How about this one?

petermattis commented on cli/cli.go:
```diff
@@ -50,9 +50,6 @@ Output build version information.
 	Use: "cockroach",
-	PersistentPreRun: func(cmd *cobra.Command, args []string) {
-		context.Addr = util.EnsureHostPort(context.Addr)
-	},
```
Ah, this is where we ensure a port on `context.Addr`. I'm curious who you didn't just add a line here to ensure a port on `context.PGAddr`. You'd have to change `EnsureHostPort` to take a port to append if one isn't present. As it is, there is a lot of duplicated functionality between `EnsureHostPort` and your new `portPadding` function.

mberhault replied:
I'm separating addr and port in
https://github.com/cockroachdb/cockroach/pull/3757
Part of that change involves proper defaults, and always concatenating host
and port.
On Jan 19, 2016 8:07 AM, "Peter Mattis" <notifications@github.com> wrote:

> In cli/cli.go
> <https://github.com/cockroachdb/cockroach/pull/3911#discussion_r50104446>:
>
> > @@ -50,9 +50,6 @@ Output build version information.
> >
> >  var cockroachCmd = &cobra.Command{
> >  	Use: "cockroach",
> > -	PersistentPreRun: func(cmd *cobra.Command, args []string) {
> > -		context.Addr = util.EnsureHostPort(context.Addr)
> > -	},
>
> Ah, this is where we ensure a port on context.Addr. I'm curious who you
> didn't just add a line here to ensure a port on context.PGAddr. You'd
> have to change EnsureHostPort to take a port to append if one isn't
> present. As it is, there is a lot of duplicated functionality between
> EnsureHostPort and your new portPadding function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/3911/files#r50104446>.
>

nak3 commented:
Thank you again @petermattis. I just hesitated to change `EnsureHostPort` to take a port to append. I updated `EnsureHostPort` and used it now.

@mberhault Although I updated here after @petermattis's review , if your PR will fix this, I'm OK with taking your patch.

tamird commented:
@mberhault can you take a look at this now that #3757 has landed?

tamird commented:
LGTM
